perfkitbenchmarker/packages/hadoop.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing Hadoop installation and cleanup functions."""
+import functools
+import logging
+import os
+import re
+
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import regex_util
+from perfkitbenchmarker import vm_util
+
+
+FLAGS = flags.FLAGS
+
+HADOOP_VERSION = '2.5.2'
+HADOOP_URL = ('http://apache.mirrors.tds.net/hadoop/common/hadoop-{0}/'
+              'hadoop-{0}.tar.gz').format(HADOOP_VERSION)
+
+DATA_FILES = ['hadoop/core-site.xml.j2', 'hadoop/yarn-site.xml.j2',
+              'hadoop/hdfs-site.xml', 'hadoop/mapred-site.xml',
+              'hadoop/hadoop-env.sh.j2', 'hadoop/slaves.j2']
+START_HADOOP_SCRIPT = 'hadoop/start-hadoop.sh.j2'
+
+HADOOP_DIR = os.path.join(vm_util.VM_TMP_DIR, 'hadoop')
+HADOOP_BIN = os.path.join(HADOOP_DIR, 'bin')
+HADOOP_SBIN = os.path.join(HADOOP_DIR, 'sbin')
+HADOOP_CONF_DIR = os.path.join(HADOOP_DIR, 'etc', 'hadoop')
+HADOOP_PRIVATE_KEY = os.path.join(HADOOP_CONF_DIR, 'hadoop_keyfile')
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  for resource in DATA_FILES + [START_HADOOP_SCRIPT]:
+    data.ResourcePath(resource)
+
+
+def _Install(vm):
+  vm.Install('openjdk7')
+  vm.RemoteCommand(('mkdir {0} && curl -L {1} | '
+                    'tar -C {0} --strip-components=1 -xzf -').format(
+                        HADOOP_DIR, HADOOP_URL))
+
+
+def YumInstall(vm):
+  """Installs Hadoop on the VM."""
+  vm.InstallPackages('snappy snappy-devel')
+  _Install(vm)
+
+
+def AptInstall(vm):
+  """Installs Hadoop on the VM."""
+  vm.InstallPackages('libsnappy1 libsnappy-dev')
+  _Install(vm)
+
+
+def _RenderConfig(vm, master_ip, worker_ips, memory_fraction=0.9):
+  yarn_memory = int((vm.total_memory_kb / 1024) * memory_fraction)
```

Comment:
That calculation seems a bit odd, usually there's fixed overhead on top of linear scaling.

I realize this is just copied from the previous version, maybe add a TODO if it's out of scope for this change?